### PR TITLE
It's ma starboard

### PR DESCRIPTION
**Pull request by Tylertron1998**
Basic starboard functionality.

### CODE REVIEW

File: events/messageReactionAdd.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+        if(!messageReaction.message.guild.configs.starboard || !messageReaction.message.guild.configs.starboardCount) return;
+        if(messageReaction.message.author.id === user.id) {
+            messageReaction.message.reply('You cannot star your own messages.')
+            .then(messageReaction.users.remove(user.id))
+            .then(message => setTimeout(() => message.delete(), 5000))
+            return;
+        }
+
+        const avatarOptions = messageReaction.message.author.displayAvatarURL().endsWith('.gif') ? { format: 'gif' } : { format: 'webp' }
```
Comment (MrJacz) [inline events/messageReactionAdd.js]: You don't need this check...

File: events/messageReactionAdd.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+const { Event } = require('klasa');
+
+module.exports = class extends Event {
+
+    constructor(...args) {
```
Comment (MrJacz) [inline events/messageReactionAdd.js]: this is what we call a useless constructor

File: events/messageReactionAdd.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            messageReaction.message.reply('You cannot star your own messages.')
+            .then(messageReaction.users.remove(user.id))
+            .then(message => setTimeout(() => message.delete(), 5000))
+            return;
+        }
+
+        const avatarOptions = messageReaction.message.author.displayAvatarURL().endsWith('.gif') ? { format: 'gif' } : { format: 'webp' }
+
+        const embed = new this.client.methods.Embed()
+        .setAuthor(messageReaction.message.member.displayName + messageReaction.message.author.discriminator, messageReaction.message.author.displayAvatarURL(avatarOptions))
```
Comment (MrJacz) [inline events/messageReactionAdd.js]: Just use `messageReaction.message.author.tag`, more unique also like I said, just use `messageReaction.message.author.displayAvatarURL()`

File: events/messageReactionAdd.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        .setDescription(messageReaction.message.content)
+        .setTimestamp()
+        .setFooter(`${messageReaction.message.id}`)
```
Comment (MrJacz) [inline events/messageReactionAdd.js]: no need for the template literals

File: events/messageReactionAdd.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+        if(!messageReaction.message.attachments.length == 0) {
```
Comment (MrJacz) [inline events/messageReactionAdd.js]: Dont use `==`, also attachments is a Collection so `messageReaction.message.attachments.size`, no need for `!` or `== 0`

File: events/messageReactionAdd.js (hunk 3 above)
Comment (Tylertron1998) [reply events/messageReactionAdd.js]: No, I want nickanames to also work, this is what I used this.

Comment (MrJacz) [reply events/messageReactionAdd.js]: @kyranet

**Comment by Soumil07** [comment]
Add the configs if they don't exist in Command.init()

**Pull request closed without merging**